### ISSUE TITLE
Module path trigger changes depending on the person running it

### ISSUE THREAD

**Issue by nitrocode** [matti/terraform-shell-resource#22]
If bob last ran this and then I ran it, I see this difference

```
-/+ resource "null_resource" "shell" {
      ~ id       = "6797389038118267585" -> (known after apply)
      ~ triggers = { # forces replacement
            "command_chomped"              = "echo hello"
            "command_when_destroy_chomped" = ":"
            "environment_keys"             = ""
            "environment_values"           = ""
          ~ "module_path"                  = "/Users/bob/git/.terraform/modules/path/matti-terraform-shell-resource-2d6c60b" -> "/Users/kyle/git/.terraform/modules/path/matti-terraform-shell-resource-2d6c60b"
            "random_uuid"                  = "f8078600-7777-787c-c5cd-0f2091af0732"
            "trigger"                      = ""
            "working_dir"                  = ""
        }
    }
```

Is there any reason for

https://github.com/matti/terraform-shell-resource/blob/master/main.tf#L8

or can we drop `abspath()` to keep this a relative path?

**Comment by matti** [comment]
see https://github.com/matti/terraform-shell-resource/pull/19#discussion_r366445282

for the reason - can we keep both cases happy?

**Comment by nitrocode** [comment]
That code is before the refactoring. I don't see how it should affect us.

Are we worried about these 2 lines?

https://github.com/matti/terraform-shell-resource/blob/master/main.tf#L37-L38

```
      "${local.module_path}/run.sh",
      local.module_path,
```

If so, we can create a new local called `absolute_path` and use that instead for those lines and keep the trigger set to `module_path` without `abspath()` function.

**Comment by matti** [comment]
to be honest I don't really know what are the side effects, but maybe we can try that and see how many people get angry. There are always older versions available.

**Comment by matti** [comment]
so, can you make the PR?

**Comment by nitrocode** [comment]
Updated the PR

**Comment by matti** [comment]
the tests still fail on this one

**Comment by matti** [comment]
https://travis-ci.org/matti/terraform-shell-resource/builds/655966408?utm_source=github_status&utm_medium=notification

**Comment by nitrocode** [comment]
Thanks. I had to install `bats-core` locally to find the issue which was a bad reference to the trigger `self.triggers.absolute_path` and had to be renamed to `local.absolute_path`. Tests are now passing.

In the future, it would be nice to run these tests in a docker container, if possible, and then document how to run locally in the `README.md`.

**Comment by matti** [comment]
In the future it would be even better to replace BATS with something that makes sense, like ruby+rspec and put that in a container. BATS is horrible, but I just dont have the time to rewrite the tests now. maybe shellspec.

**Comment by nitrocode** [comment]
really? It's my first time using it but I thought it was kind of nifty 😄

**Comment by matti** [comment]
Yes it's nifty after I wrote all this https://github.com/matti/terraform-shell-resource/blob/master/tests/helper.bash

Like this https://shellspec.info looks actually good.

(continuing the actual discussion in #23 )

**Comment by matti** [comment]
fixed in v1.0.3, thanks @nitrocode
